code/modules/economy/atm.dm

```diff
@@ -0,0 +1,230 @@
+						dat += "[selected ? "<span class='linkOff'>" : "<a href='?src=\ref[src];selectaccount=[acc.name]'>"][acc.name][acc.owner ? " ([acc.owner])" : ""][acc.verified ? " <font color='green'>V</font>" : ""] [acc.balance] [CURRENCY(acc.balance)][selected ? "</span>" : "</a>"]\n"
+
+					dat += 	"<hr>"
+					dat += 	"<form name='search' action='?src=\ref[src]'>\
+										<input type='hidden' name='src' value='\ref[src]'>\
+										<input type='hidden' name='search' value='tosearch'>\
+										<input type='text' name='tosearch' value='[tosearch]'>\
+										<input type='submit' value='Search'>\
+										</form>\n\n"
+
+					for(var/datum/bankaccount/acc in bank.accounts)
+						var/pendingcontent = "[acc.name][acc.owner ? " ([acc.owner])" : ""]"
+						if(tosearch && findtext(pendingcontent, tosearch))
+							dat += "[selected ? "<span class='linkOff'>" : "<a href='?src=\ref[src];selectaccount2=[acc.name]'>"][pendingcontent][acc.verified ? " <font color='green'>V</font>" : ""] [acc.balance] [CURRENCY(acc.balance)][selected ? "</span>" : "</a>"]\n"
+
+				if(5) //Manage accounts
+					if(!selectedaccount)
+						dat += "Please select an account to begin."
+						
+						for(var/datum/bankaccount/acc in card.accountlist)
+							dat += "<a href='?src=\ref[src];selectaccount=[acc.name]'>[acc.name][acc.owner ? " ([acc.owner])" : ""][acc.verified ? " <font color='green'>V</font>" : ""] [acc.balance] [CURRENCY(acc.balance)]</a>\n"
+
+					else if(selectedaccount in card.accountlist) //cant have people looking at other people's account details!
+						dat += "<a href='?src=\ref[src];selectaccount=0'>Back</a>\n"
+						dat += "<hr>"
+						dat += ""
+						dat += "Account number:\t[selectedaccount.name]\n"
+						dat += "Account owner:\t[selectedaccount.verified ? "<span class='LinkOff'>" : "<a href='?src=\ref[src];changename=1'>"][selectedaccount.owner][selectaccount.verified ? "</span>" : "</a>"]\n"
+						dat += "Balance:\t[selectedaccount.balance]\n"
+						dat += "Verified: \t[selectedaccount.verified ? "Yes" : "No"]\n"
+						dat += "Linked Cards: t[selectedaccount.cards.len]\n"
+						
+						dat += "<hr>"
+						dat += "<b>Logs</b>\n"
+						for(var/log in selectaccount.logs)
+							dat += log
+							dat += "\n"
+
+				if(6 to INFINITY) //holy shit i hate making uis
+					dat += "ayy lmao"
+
+	popup.set_content(dat)
+	popup.open()
+
+/obj/machinery/atm/Topic(href, href_list)
+	if(..() || !bank.isOperational())
+		return
+
+	if(href_list["changemode"])
+		mode = href_list["changemode"]
+		updateDialog()
+		return
+
+	if(href_list["selectaccount"])
+		selectedaccount = bank.accounts[href_list["selectedaccount"]]
+		updateDialog()
+		return
+	
+	if(href_list["selectaccount2"])
+		selectedaccount2 = bank.accounts[href_list["selectedaccount2"]]
+		updateDialog()
+		return
+	
+	if(href_list["search"])
+		var/tosearch = href_list["tosearch"]
+		updateDialog()
+		return
+
+	if(href_list["depositcash"])
+		var/value = 0
```

Comment:
i should probably put these in their own procs lest i invoke the wrath of gia's ghost